robosat/tools/tile.py

```diff
@@ -0,0 +1,123 @@
+    parser.add_argument("--zoom", type=int, required=True, help="zoom level of tiles")
+    parser.add_argument("--type", type=str, default="image", help="image or label tiling")
+    parser.add_argument("--dataset", type=str, help="path to dataset configuration file, mandatory for label tiling")
+    parser.add_argument("--no_data", type=int, help="color considered as no data [0-255]. Skip related tile")
+    parser.add_argument("--leaflet", type=str, help="leaflet client base url")
+
+    parser.set_defaults(func=main)
+
+
+def main(args):
+
+    if args.type == "label":
+        try:
+            dataset = load_config(args.dataset)
+        except:
+            sys.exit("Error: Unable to load DataSet config file")
+
+        classes = dataset["common"]["classes"]
+        colors = dataset["common"]["colors"]
+        assert len(classes) == len(colors), "classes and colors coincide"
+        assert len(colors) == 2, "only binary models supported right now"
+
+    try:
+        raster = rasterio_open(args.raster)
+        w, s, e, n = bounds = transform_bounds(raster.crs, "EPSG:4326", *raster.bounds)
+        transform, _, _ = calculate_default_transform(raster.crs, "EPSG:3857", raster.width, raster.height, *bounds)
+    except:
+        sys.exit("Error: Unable to load raster or deal with it's projection")
+
+    tiles = [mercantile.Tile(x=x, y=y, z=z) for x, y, z in mercantile.tiles(w, s, e, n, args.zoom)]
+    tiles_nodata = []
+
+    for tile in tqdm(tiles, desc="Tiling", unit="tile", ascii=True):
+
+        w, s, e, n = tile_bounds = mercantile.xy_bounds(tile)
+
+        # Inspired by Rio-Tiler, cf: https://github.com/mapbox/rio-tiler/pull/45
+        warp_vrt = WarpedVRT(
+            raster,
+            crs="EPSG:3857",
+            resampling=Resampling.bilinear,
+            add_alpha=False,
+            transform=from_bounds(*tile_bounds, args.size, args.size),
+            width=math.ceil((e - w) / transform.a),
+            height=math.ceil((s - n) / transform.e),
+        )
+        data = warp_vrt.read(out_shape=(len(raster.indexes), args.size, args.size), window=warp_vrt.window(w, s, e, n))
+
+        # If no_data is set, remove all tiles with at least one whole border filled only with no_data (on all bands)
+        if type(args.no_data) is not None and (
+            np.all(data[:, 0, :] == args.no_data)
+            or np.all(data[:, -1, :] == args.no_data)
+            or np.all(data[:, :, 0] == args.no_data)
+            or np.all(data[:, :, -1] == args.no_data)
+        ):
+            tiles_nodata.append(tile)
+            continue
+
+        C, W, H = data.shape
+
+        os.makedirs(os.path.join(args.out, str(args.zoom), str(tile.x)), exist_ok=True)
+        path = os.path.join(args.out, str(args.zoom), str(tile.x), str(tile.y))
+
+        if args.type == "label":
+            assert C == 1, "Error: Label raster input should be 1 band"
+
+            ext = "png"
```

Comment:
Should this be configurable?

Reply:
Not sure to understand what do you have in mind on this one,